web-components/src/components/dropdown/Dropdown.ts

```diff
@@ -0,0 +1,476 @@
+          this.focusedIndex++;
+        } else {
+          this.focusFirst();
+        }
+      }
+    }
+
+    focusPrev() {
+      if (this.renderOptions.length) {
+        if (this.focusedIndex > 0) {
+          this.focusedIndex--;
+        } else {
+          this.focusLast();
+        }
+      }
+    }
+
+    focusToIndex(n: number) {
+      if (this.renderOptions.length) {
+        if (n >= 0 && n <= this.renderOptions.length - 1) {
+          this.focusedIndex = n;
+        }
+      }
+    }
+
+    focusReset() {
+      this.focusedIndex = -1;
+    }
+
+    async scrollToIndex(n: number) {
+      await new Promise<void>(resolve => {
+        requestAnimationFrame(() => {
+          if (
+            this.optionsListItems &&
+            this.optionsListItems.length > this.visibleOptions &&
+            n >= 0 &&
+            this.optionsListItems.length > n
+          ) {
+            let distance = 0;
+
+            const { top, bottom } = this.optionsList.getBoundingClientRect();
+            const option = this.optionsListItems[n];
+
+            const nextOption = this.optionsListItems[n + 1] || option;
+            const prevOption = this.optionsListItems[n - 1] || option;
+
+            const nextOptionRect = nextOption.getBoundingClientRect();
+            const prevOptionRect = prevOption.getBoundingClientRect();
+
+            if (nextOptionRect.bottom > bottom) {
+              distance = nextOptionRect.bottom - bottom + 2;
+            } else if (prevOptionRect.top < top) {
+              distance = prevOptionRect.top - top - 2;
+            }
+
+            this.optionsList.scrollTop += distance;
+          }
+          resolve();
+        });
+      });
+    }
+
+    getOptionKeyValuePair(option: Option) {
+      if (typeof option === "string") {
+        return { key: option.trim(), value: option };
+      }
+
+      if (option && typeof option === "object") {
+        const optionKeys = Object.keys(option as OptionMember);
+        const optionValues = Object.values(option as OptionMember);
+
+        if (optionKeys.length) {
+          if (optionKeys.length === 1 || !this.optionId) {
+            return { key: optionKeys[0], value: optionValues[0] };
+          } else if (this.optionId) {
+            return { key: option[this.optionId], value: option[this.optionValue || this.optionId] };
+          }
+        }
+      }
+
+      return { key: "undefined", value: "undefined" };
+    }
+
+    get labelTitle() {
+      if (this.selectedKey) {
+        const option = this.renderOptions.find(o => o.key === this.selectedKey);
+        if (option) {
+          return option.value;
+        }
+      }
+      return this.title;
+    }
+
+    get dropDownClassMap() {
+      return {
+        "md-dropdown__expanded": this.expanded
+      };
+    }
+
+    render() {
+      return html`
+        <div class="md-dropdown ${classMap(this.dropDownClassMap)}" part="dropdown">
+          <label
```

Comment:
With Mac VOice Over - it announces : "One Group", "24 Group" instead of "One, Choose an option, you are currently on a collapsed popup button" like here: https://www.w3schools.com/tags/tryit.asp?filename=tryhtml_select